### PR TITLE
update Moderation example to best practice and v4

### PR DESCRIPTION
**Pull request by drbenvincent**
Updates the moderation analysis example (see #170) to current best practice.

Because this is just a minor update with no substantive changes, I propose to speed things up that we don't necessarily need 2 reviewers. The only meaningful change was to shift all custom plot functions into a single (hidden) code cell.

Note: in `pm.sample`, I'm currently having to provide `nuts={"target_accept": 0.9}` rather than just `target_accept=0.9` (see https://github.com/pymc-devs/pymc/issues/5620)

### CODE REVIEW

File: examples/references.bib [hunk 1]
```diff
@@ -302,6 +358,13 @@ @misc{szegedy2014going
   archiveprefix = {arXiv},
   primaryclass  = {cs.CV}
 }
+@online{vandenbergSPSS,
+  author        = {van den Berg, R. G},
+  title         = {SPSS Moderation Regression Tutorial},
+  year          = XXXXX,
```
Comment (OriolAbril) [inline examples/references.bib:364]: I think this is breaking the docs build. Can you try removing the year key altogether?

File: myst_nbs/case_studies/moderation_analysis.myst.md [hunk 2]
```diff
@@ -6,27 +6,32 @@ jupytext:
...
+  name: pymc-dev-py39
 ---
 
+(moderation_analysis)=
 # Bayesian moderation analysis
 
-**Author:** [Benjamin T. Vincent](https://github.com/drbenvincent)
+:::{post} March, 2022
+:tags: moderation, path analysis,
+:category: beginner
```
Comment (OriolAbril) [inline myst_nbs/case_studies/moderation_analysis.myst.md:19]:
```suggestion
:category: beginner, explanation
```

I think this notebook fits relatively well the explanation concept, especially now that auxiliary plotting is hidden.

**Comment by drbenvincent** [comment]
<p>done</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/297/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by drbenvincent** [comment]
<p>have removed coords</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/297/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by drbenvincent** [comment]
It's not adding the `pm.*` tags.

But docs now seems to work :)

**Comment by drbenvincent** [comment]
Everything looks good, _but_ still no `pm.*` tags

**Comment by OriolAbril** [comment]
> Everything looks good, but still no pm.* tags

I think we still need to update the style guide, but you can see that CI is passing without them. They should not be present anymore (and should be removed from the ones that have them).

**Comment by drbenvincent** [comment]
<p>Have attempted to cross reference the two notebooks</p>

---

 <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/297/discussion/'>View entire conversation</a> on ReviewNB<div id="RAICReplyForNotification-ReviewNBCommentContext-DoNotDelete"></div>

**Comment by drbenvincent** [comment]
Notebook cross references now work fine

**Review by OriolAbril — APPROVED** [review]

**Comment by OriolAbril** [comment]
Thanks! How has the extra myst step in pre-commit worked out for you?

**Comment by drbenvincent** [comment]
For these kinds of small changes I don't think there was much difference. But I think it will definitely be more useful with larger changes. Better diff.

I was having some issues when running the pre commit on a target notebook: 

Screenshot 2022-03-20 at 17 44 19

I ended up running `pre-commit run --all`

**Comment by OriolAbril** [comment]
I have never tried that on specific files and the error doesn't make much sense to me, could it be because it was unstaged (before doing git add that then shows in git status in green)?

**Pull request merged**